exporters/fluentd/test/log/fluentd_recordable_test.cc

```diff
@@ -0,0 +1,188 @@
+template <typename timeunit> void yield_for(timeunit duration) {
+  auto start = std::chrono::high_resolution_clock::now();
+  auto end = start + duration;
+  do {
+    std::this_thread::yield();
+  } while (std::chrono::high_resolution_clock::now() < end);
+}
+
+using Properties = std::map<std::string, opentelemetry::common::AttributeValue>;
+
+struct TestServer {
+  SocketServer &server;
+  std::atomic<uint32_t> count{0};
+
+  TestServer(SocketServer &server) : server(server) {
+    server.onRequest = [&](SocketServer::Connection &conn) {
+      std::vector<uint8_t> msg(conn.request_buffer.data(),
+                               conn.request_buffer.data() +
+                                   conn.request_buffer.size());
+
+      try {
+        auto j = nlohmann::json::from_msgpack(msg);
+        std::cout << "[" << count.fetch_add(1)
+                  << "] SocketServer received payload: " << std::endl
+                  << j.dump(2) << std::endl;
+
+        conn.response_buffer = j.dump(2);
+        conn.state.insert(SocketServer::Connection::Responding);
+        conn.request_buffer.clear();
+      } catch (std::exception &) {
+        conn.state.insert(SocketServer::Connection::Receiving);
+        // skip invalid payload
+      }
+    };
+  }
+
+  void Start() { server.Start(); }
+
+  void Stop() { server.Stop(); }
+
+  void WaitForEvents(uint32_t expectedCount, uint32_t timeout) {
+    if (count.load() != expectedCount) {
+      yield_for(std::chrono::milliseconds(timeout));
+    }
+    EXPECT_EQ(count.load(), expectedCount);
+  }
+};
+
+TEST(FluentdExporter, SendLogEvents) {
+  bool isRunning = true;
+
+  // Start test server
+  SocketAddr destination("127.0.0.1:24222");
+  SocketParams params{AF_INET, SOCK_STREAM, 0};
+  SocketServer socketServer(destination, params);
+  TestServer testServer(socketServer);
+  testServer.Start();
+
+  yield_for(std::chrono::milliseconds(500));
+
+  // Connect to local test server
+  opentelemetry::exporter::fluentd::common::FluentdExporterOptions options;
+  options.endpoint = "tcp://127.0.0.1:24222";
+  options.tag = "tag.my_service";
+
+  auto exporter = std::unique_ptr<sdklogs::LogExporter>(
+      new opentelemetry::exporter::fluentd::logs::FluentdExporter(options));
+  auto processor = std::shared_ptr<sdklogs::LogProcessor>(
+      new sdklogs::SimpleLogProcessor(std::move(exporter)));
+
+  auto provider = std::shared_ptr<opentelemetry::logs::LoggerProvider>(
+      new opentelemetry::sdk::logs::LoggerProvider());
+
+  auto pr =
+      static_cast<opentelemetry::sdk::logs::LoggerProvider *>(provider.get());
+
+  pr->SetProcessor(processor);
+
+  // Set the global trace provider
+  opentelemetry::logs::Provider::SetLoggerProvider(provider);
+
+  std::string providerName = "MyInstrumentationName";
+  auto logger = provider->GetLogger(providerName);
+
+  // Span attributes
+  // Properties attribs = {{"attrib1", 1}, {"attrib2", 2}};
+
+  logger->Log(logs::Severity::kDebug, "f2");
+  /*{
```

Comment:
Could this commented block be just removed?

Reply:
Yes, it shouldn't be there. Have removed now.